### DISCUSSION TITLE
Grain Growth - grain_growth_2D_random.i

### DISCUSSION THREAD

**Discussion by EtienneSwerim** [idaholab/moose#16966] [Q&A Modules: Phase field] [answered]
Hi,

Could I get a little help with the appended example file: grain_growth_2D_random.i?

As I understand from the Grain Growth Model section of the Moose Manual, order parameters and kernels are automatically created by the Action system called GrainGrowthAction. I even found the files GrainGrowthAction.h and GrainGrowthAction.C and read them.

What I do not understand is:

1) Where, in the grain_growth_2D_random.i, are the GrainGrowthAction files called? The name GrainGrowthAction is not present in the file! I guess these files are called implicitly but how?

2) When it comes to variables, only the name PolycrystalVariables is defined but neither a type nor an action, see below.
```
[Variables]
  # Variable block, where all variables in the simulation are declared
  [./PolycrystalVariables]
    # Custom action that created all of the grain variables and sets their initial condition
  [../]
[]
```
 I noticed that Moose crashes if the name PolycrystalVariables is changed and I noticed that there is a folder called PolycrystalVariables
(moose/module/phase_field/doc/content/syntax/Variables/PolycrystalVariables)

This folder contains a single file called index.md which contain the following single line:

!template load file=stubs/moose_system.md.template name=PolycrystalVariables syntax=/Variables/PolycrystalVariables

a) Is this index.md file called by the variable name PolycrystalVariables in the grain_growth_2D_random.i file?

b) What does this index.md file do? I noticed there were a lot of theses files in the moose folder for example one below PolycrystalKernel and one below PolycrystalRandomIC and I do not understand what these two do either.

Very grateful for help!

Best Regards
Etienne

[grain_growth_2D_random.txt](https://github.com/idaholab/moose/files/5968379/grain_growth_2D_random.txt)

**Comment by SudiptaBiswas** [comment, marked as answer]
The input file you are referring to doesn't use `GrainGrowthAction`. Instead, it separately uses the `PolycrystalVariableAction` and `PolycrystalkernelAction`.  `Variables/PolycrystalVariables` calls for the action that sets up the variables based on `variable_base_name` and `op_num`. You can find out how the actions are registered in the `PhaseFieldApp.C` file. 

`*.md` files are the documentation files. You will find more documentation on this on the mooseframework website.

**Reply by EtienneSwerim** [reply]
Hi Sudipta,

Thanks a lot for your answer!
Very useful to know about the PhaseFieldApp.C file.
As I understand from your answer, when the input file contains:
[Variables]
  [./PolycrystalVariables]
  [../]
[]
Moose calls the files: PolycrystalVariablesAction.h and PolycrystalVariablesAction.C.

And good to know that all the .md files are for documentation and not calculation.

Again, thanks a lot! I am new to Moose so help is greatly appreciated.

Best Regards

Etienne
